src/main/kotlin/app/revanced/patches/music/misc/translations/TranslationsPatch.kt

```diff
@@ -1,46 +1,87 @@
 package app.revanced.patches.music.misc.translations
 
 import app.revanced.patcher.data.ResourceContext
+import app.revanced.patcher.patch.PatchException
+import app.revanced.patcher.patch.options.PatchOption.PatchExtensions.stringPatchOption
 import app.revanced.patches.music.utils.compatibility.Constants.COMPATIBLE_PACKAGE
 import app.revanced.patches.music.utils.settings.SettingsPatch
+import app.revanced.patches.shared.translations.APP_LANGUAGES
 import app.revanced.patches.shared.translations.TranslationsUtils.copyXml
+import app.revanced.patches.shared.translations.TranslationsUtils.updateStringsXml
 import app.revanced.util.patch.BaseResourcePatch
+import java.io.File
 
-@Suppress("unused")
+// Array of supported RVX languages, each represented by its language code.
+val LANGUAGES = arrayOf(
+    "bg-rBG", "bn", "cs-rCZ", "el-rGR", "es-rES", "fr-rFR", "hu-rHU", "id-rID", "in", "it-rIT",
+    "ja-rJP", "ko-rKR", "nl-rNL", "pl-rPL", "pt-rBR", "ro-rRO", "ru-rRU", "tr-rTR", "uk-rUA",
+    "vi-rVN", "zh-rCN", "zh-rTW"
+)
+
+@Suppress("DEPRECATION", "unused")
 object TranslationsPatch : BaseResourcePatch(
     name = "Translations",
     description = "Adds Crowdin translations for YouTube Music.",
     dependencies = setOf(SettingsPatch::class),
     compatiblePackages = COMPATIBLE_PACKAGE
 ) {
+    private var CustomLanguage by stringPatchOption(
+        key = "CustomLanguage",
+        default = "",
+        title = "Custom language file",
+        description = """
+            The file path to the strings.xml file.
+            Please note that applying the strings.xml file will overwrite all existing language translations.
+            """.trimIndent()
+    )
+
+    private var SelectedLanguages by stringPatchOption(
+        key = "SelectedLanguages",
+        default = LANGUAGES.joinToString(", "),
+        title = "RVX languages to add",
+        description = "A list of languages to be added for the RVX settings, separated by commas."
+    )
+
+    private var SelectedAppLanguages by stringPatchOption(
+        key = "SelectedAppLanguages",
+        default = APP_LANGUAGES.joinToString(", "),
+        title = "App languages to keep",
+        description = "A list of languages to be kept for the app, separated by commas. Languages not in the list will be removed from the app."
```

Comment:
This feature removes the default string resource for YouTube or YouTube Music
Is my understanding correct?

If so, wouldn't it make sense to separate it into different patches?

Because the description of the `Translations` patch is:
```
Adds Crowdin translations for YouTube Music
```

Reply:
How about changing the patch description to something like: `Adds or changes the languages in the app.` or `Adds or changes the languages in the app using Crowdin.`

Reply:
Oh yeah, I never thought there was such a simple way

Reply:
Maybe it's also worth noting in the description that English is the default language and will never be removed. Some users have encountered difficulties understanding this and kept `en-*` for English only (e.g., en-uk, en-au). However, to ensure only English (US) remains, the list should be empty.

These options for RVX and YT/YTM languages reduce the app size. It might be beneficial to mention this for less tech-savvy individuals, as I've observed people questioning the necessity of these options.

Reply:
> Maybe it's also worth noting in the description that English is the default language and will never be removed. Some users have encountered difficulties understanding this and kept `en-*` for English only (e.g., en-uk, en-au). However, to ensure only English (US) remains, the list should be empty.

Is the default English one known as `en-us` or does that term not exist since it is the core strings.xml?

What I mean to ask is which of the following potential descriptions is correct:

A: `A list of languages to be kept for the app, separated by commas. Except for English, languages not in the list will be removed from the app.

B: `A list of languages to be kept for the app, separated by commas. Except for en-us, languages not in the list will be removed from the app.

Reply:
The first one is correct. Default English does not have language code. Even if they delete all language codes, including all `en-*`, English will always be there.